File: leetcode-python-go/1779. 找到最近的有相同 X 或 Y 坐标的点/main.go
package main

import (
	"fmt"
	"math"
)

func nearestValidPoint(x, y int, points [][]int) int {
	ans := -1
	minDist := math.MaxInt32
	for i, p := range points {
		if p[0] == x || p[1] == y {
			dist := abs(p[0]-x) + abs(p[1]-y)
			if dist < minDist {
				minDist = dist
				ans = i
			}
		}
	}
	return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func main() {
	x := 3
	y := 4
	points := [][]int{{1, 2}, {3, 1}, {2, 4}, {2, 3}, {4, 4}}
	results := nearestValidPoint(x, y, points)
	fmt.Println(results)
}

File: leetcode-python-go/1779. 找到最近的有相同 X 或 Y 坐标的点/main.py
from typing import List

x = 3
y = 4
points = [[1,2],[3,1],[2,4],[2,3],[4,4]]

def nearestValidPoint(x: int, y: int, points: List[List[int]]) -> int:
    n = len(points)
    best, bestid = float("inf"), -1
    for i, (px, py) in enumerate(points):
        if x == px:
            if (dist := abs(y - py)) < best:
                best = dist
                bestid = i
        elif y == py:
            if (dist := abs(x - px)) < best:
                best = dist
                bestid = i
    
    return bestid

print(nearestValidPoint(x, y, points))